packages/zowe-explorer/src/webviews/src/table-view/ActionsBar.tsx

```diff
@@ -1,66 +1,174 @@
-import { VSCodeButton } from "@vscode/webview-ui-toolkit/react";
+import { VSCodeButton, VSCodeTextField } from "@vscode/webview-ui-toolkit/react";
 import { GridApi } from "ag-grid-community";
+import { wrapFn } from "./types";
+import { FocusableItem, Menu, MenuGroup, MenuItem } from "@szhsin/react-menu";
+import "@szhsin/react-menu/dist/index.css";
 
 export const ActionsBar = ({
   actions,
   gridRef,
   itemCount,
+  selectionCount,
+  title,
+  columns,
+  visibleColumns,
+  setVisibleColumns,
   vscodeApi,
 }: {
   actions: Table.Action[];
   gridRef: Ref<any>;
   itemCount: number;
+  selectionCount: number;
+  title: string;
+  columns: string[];
+  visibleColumns: string[];
+  setVisibleColumns: Dispatch<string[]>;
   vscodeApi: any;
 }) => {
+  const [searchFilter, setSearchFilter] = useState<string>("");
+
+  const columnDropdownItems = (visibleColumns: string[]) =>
+    columns
+      .filter((col) => col !== "actions" && (searchFilter.length === 0 || col.toLowerCase().includes(searchFilter)))
+      .map((col) => (
+        <MenuItem
+          key={`toggle-vis-${col}`}
+          type="checkbox"
+          onClick={(e: any) => {
+            const gridApi = gridRef.current.api as GridApi;
+            const colVisibility = !visibleColumns.includes(col);
+            gridApi.setColumnsVisible(
+              [gridApi.getColumns()?.find((c) => c.getColDef().field === col || c.getColDef().headerName === col)!],
+              colVisibility
+            );
+            setVisibleColumns(colVisibility ? [...visibleColumns, col] : visibleColumns.filter((c) => c !== col));
+            e.keepOpen = true;
+          }}
+        >
+          {() => (
+            <div
+              style={{
+                display: "flex",
+                justifyContent: "space-evenly",
+                alignItems: "center",
+                height: "1.25rem",
+                marginLeft: "-1.25rem",
+                overflow: "hidden",
+              }}
+            >
+              <span style={{ paddingRight: visibleColumns.includes(col) ? "1em" : "2.15em", marginTop: "3px" }}>
+                {visibleColumns.includes(col) ? <span className="codicon codicon-check"></span> : null}
+              </span>
+              {col}
+            </div>
+          )}
+        </MenuItem>
+      ));
+
   return (
     <div
       style={{
-        height: "2.5em",
+        height: "3em",
         display: "flex",
         alignItems: "center",
-        borderRadius: "var(--ag-wrapper-border-radius)",
         border: "1px solid var(--vscode-editorWidget-border)",
+        borderTopLeftRadius: "var(--ag-wrapper-border-radius)",
+        borderTopRightRadius: "var(--ag-wrapper-border-radius)",
         justifyContent: "space-between",
         backgroundColor: "var(--vscode-keybindingTable-headerBackground)",
         color: "var(--vscode-foreground) !important",
         padding: "0 0.25em",
-        marginBottom: "3px",
+        marginBottom: "-1px",
```

Comment:
same as previous comment 😅

Reply:
Agreed, if we need to use this elsewhere we can pull it out into a CSS class. Since its only used for the mentioned component, I think we can leave it for now.